connectors/sources/confluence.py

```diff
@@ -984,8 +992,9 @@ async def search_by_query(self, query):
             # entity can be space or content
             entity_details = entity.get(SPACE) or entity.get(CONTENT)
 
-            if (
-                entity_details.get("type", "") == "attachment"
+            if not entity_details:
+                continue
+            if (entity_details.get("type", "") == "attachment"
```

Comment:
```suggestion
            if not entity_details or (entity_details.get("type", "") == "attachment"
```

might be a bit cleaner. But what's the scenario you found where `entity_details` does not exist?

Reply:
I never debugged that into depth, I just noticed that the connector sometimes crashes because of entity_details not being around. I can look into this closer